components/event/org.wso2.carbon.event.admin/src/main/java/org/wso2/carbon/event/admin/internal/TopicManagerAdminService.java

```diff
@@ -82,56 +137,110 @@ public void updatePermission(String topic, TopicRolePermission[] topicRolePermis
         try {
             return adaptSubscriptions(eventBroker.getTopicManager().getJMSSubscriptions(topic));
         } catch (EventBrokerException e) {
-            throw new EventAdminException("Can not get the jms subscriptions", e);
+            log.error("Cannot get the jms subscriptions", e);
+            throw new EventAdminException("Cannot get the jms subscriptions", e);
         }
     }
 
-    private Subscription adaptSubscription(
-            org.wso2.carbon.event.core.subscription.Subscription coreSubscription) {
-        Calendar calendar = Calendar.getInstance();
-        calendar.setTime(coreSubscription.getCreatedTime());
-        Subscription adminSubscription = new Subscription();
-        adminSubscription.setCreatedTime(calendar);
-        adminSubscription.setEventDispatcher(coreSubscription.getEventDispatcher());
-        adminSubscription.setEventDispatcherName(coreSubscription.getEventDispatcherName());
-        adminSubscription.setEventFilter(coreSubscription.getEventFilter());
-        adminSubscription.setEventSinkURL(coreSubscription.getEventSinkURL());
-        adminSubscription.setExpires(coreSubscription.getExpires());
-        adminSubscription.setId(coreSubscription.getId());
-        adminSubscription.setOwner(coreSubscription.getOwner());
-        adminSubscription.setTopicName(coreSubscription.getTopicName());
-        adminSubscription.setMode(coreSubscription.getMode());
-        return adminSubscription;
+    /**
+     * Gets user roles through topic manager
+     *
+     * @return A string array of roles
+     * @throws EventAdminException
+     */
+    public String[] getUserRoles() throws EventAdminException {
+        EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
+        try {
+            return eventBroker.getTopicManager().getBackendRoles();
+        } catch (EventBrokerException e) {
+            log.error("Error in getting user roles from topic manager", e);
+            throw new EventAdminException("Error in getting user roles from topic manager", e);
+        }
+    }
+
+    /**
+     * Removes a topic
+     * Suppressing warning as this is used as a web service
+     *
+     * @param topic Topic name
+     * @return true if topic existed to delete and deleted, false otherwise.
+     * @throws EventAdminException
+     */
+    @SuppressWarnings("UnusedDeclaration")
+    public boolean removeTopic(String topic) throws EventAdminException {
+        EventBroker eventBroker = EventAdminHolder.getInstance().getEventBroker();
+        try {
+            return eventBroker.getTopicManager().removeTopic(topic);
+        } catch (EventBrokerException e) {
+            log.error("Error in removing a topic", e);
+            throw new EventAdminException("Error in removing a topic", e);
+        }
     }
 
+    /**
+     * Converting carbon event core subscription array to carbon event internal subscription array
```

Comment:
This is within the margin of 100 columns. I.E Right margin (columns) is set to 100.